client-sdk/go/types/transaction.go

```diff
@@ -56,35 +79,61 @@ type TransactionSigner struct {
 	ut UnverifiedTransaction
 }
 
+func (ts *TransactionSigner) allocateProofs() {
+	if len(ts.ut.AuthProofs) == 0 {
+		ts.ut.AuthProofs = make([]AuthProof, len(ts.tx.AuthInfo.SignerInfo))
+
+		for i, si := range ts.tx.AuthInfo.SignerInfo {
+			if si.AddressSpec.Multisig != nil {
+				if len(ts.ut.AuthProofs[i].Multisig) == 0 {
+					ts.ut.AuthProofs[i].Multisig = make([][]byte, len(si.AddressSpec.Multisig.Signers))
+				}
+			}
+		}
+	}
+}
+
 // AppendSign signs the transaction and appends the signature.
 //
 // The signer must be specified in the AuthInfo.
 func (ts *TransactionSigner) AppendSign(ctx signature.Context, signer signature.Signer) error {
 	pk := signer.Public()
-	index := -1
+	any := false
 	for i, si := range ts.tx.AuthInfo.SignerInfo {
-		if !si.PublicKey.Equal(pk) {
-			continue
-		}
+		switch {
```

Comment:
Oh I see, so now you sign all the slots that correspond to the signer.

Reply:
yeah :shrug: the typescript client is different in style